hase: init at unstable-2020-10-06

uninsane opened the pull request:
project page: <http://ziz.gp2x.de/hase/index.htm>

"Hase is an open source (GPL) gravity based artillery shooter. It is similar to Worms, Hedgewars or artillery, but the gravity force and direction depends on the mass nearby."

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+    pkg-config
+  ];
+
+  propagatedBuildInputs = [
```
```suggestion
  buildInputs = [
```

uninsane replied:
changing `propagatedBuildInputs` to `buildInputs` here causes `nix buld '.#hase'` to fail with:
```
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Package sdl was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sdl', required by 'sparrow3d', not found
```

sparrow both links to sdl and when a project like Hase `#include <sparrow3d.h>`s that in turn `#include <SDL.h>`s. i think `propagatedBuildInputs` is the solution here, though maybe the sparrow3d.pc pkg-config file could `-Isystem ${sdl.dev}/include/SDL` & friends as an alternative, if you think that's better?

AndersonTorres replied:
Well, two things:

1. I believe you can use the function `makePkgconfigItem` to generate pkg-config `.pc` files;
2. I don't know how other distros handle things like this. _However_, given that other distros just throw everything in a global namespace, they are not useful to draw a line here.

OK, keep `propagated???`

uninsane replied:
switched to `makePkgconfigItem` (thanks for letting me know about it!) & kept `propagatedBuildInputs`

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+  ];
+
+  postConfigure = ''
+    # sparrow expects the SDL_* libraries to already be on the ld path
```
```suggestion
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+    NIX_CFLAGS_COMPILE=$(pkg-config --cflags SDL_image SDL_ttf SDL_mixer SDL_net)
+  '';
+
+  buildPhase = ''
+    make dynamic
+  '';
```
```suggestion
  buildFlags = [ "dynamic" ];
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+    make dynamic
+  '';
+
+  installPhase = ''
+    # `dev` output
+    mkdir -p $dev/{include,lib/pkgconfig}
+    cp sparrow*.h $dev/include
+    substituteAll ${./sparrow3d.pc.in} $dev/lib/pkgconfig/sparrow3d.pc
+
+    # `out` output
+    mkdir -p $out/lib
+    cp libsparrow{3d,Net,Sound}.so $out/lib
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/lib
    cp libsparrow{3d,Net,Sound}.so $out/lib

    mkdir -p $dev/{include,lib/pkgconfig}
    cp sparrow*.h $dev/include
    substituteAll ${./sparrow3d.pc.in} $dev/lib/pkgconfig/sparrow3d.pc

    runHook postInstall
  '';
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+    cp libsparrow{3d,Net,Sound}.so $out/lib
+  '';
+
+  checkPhase = ''
+    make all_no_static
+    ./testfile.sh
+    # the other test cases are interactive
+    # ./testsparrow
+  '';
```
```suggestion
  doCheck = true;
  
  checkPhase = ''
    runHook preCheck
  
    make all_no_static
    ./testfile.sh
    
    runHook postCheck  
  '';
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+    # ./testsparrow
+  '';
+
+  doCheck = true;
```
```suggestion
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,73 @@
+
+  doCheck = true;
+
+  meta = with lib; {
+    description = "a software renderer for different open handhelds like the gp2x, wiz, caanoo and pandora";
+    homepage = "https://github.com/theZiz/sparrow3d";
+    license = licenses.lgpl21;
+    maintainers = with maintainers; [ colinsane ];
+    platforms = platforms.linux;
+  };
```
```suggestion
  meta = {
    homepage = "https://github.com/theZiz/sparrow3d";
    description = "A software renderer for different open handhelds like the gp2x, wiz, caanoo and pandora";
    license = lib.licenses.lgpl21;
    maintainers = with lib.maintainers; [ colinsane ];
    platforms = platforms.linux;
  };
```

AndersonTorres commented on pkgs/development/libraries/sparrow3d/sparrow3d.pc.in:
```diff
@@ -0,0 +1,17 @@
+  SDL_net
+Cflags: -isystem${includedir}
+Libs: -L${libdir} -lsparrow3d -lsparrowNet -lsparrowSound
+
```
```suggestion
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+    rev = "31d6840cdf0c72fc459f10402dae7726096b2974";
+    hash = "sha256-d9So3E8nCQJ1/BdlwMkGbaFPT9mkX1VzlDGKp71ptEE=";
+  };
+  patches = [ ./prefer-dynamic.patch ];
```
```suggestion

  patches = [ ./prefer-dynamic.patch ];
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+  ];
+
+  buildPhase = ''
+    # hase expects its dependencies to already be on the ld path
```
```suggestion
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+  '';
+
+  meta = with lib; {
+    description = "Hase is an open source gravity based artillery shooter. It is similar to Worms, Hedgewars or artillery, but the gravity force and direction depends on the mass nearby. It is optimized for mobile game consoles like the GP2X, Open Pandora or GCW Zero";
```
```suggestion
    description = "An open-source artillery shooter";
    longDescription = ''
      Hase is an open source gravity based artillery shooter. It is similar to
      Worms, Hedgewars or artillery, but the gravity force and direction
      depends on the mass nearby. It is optimized for mobile game consoles like 
      the GP2X, Open Pandora or GCW Zero.
    '';
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+      --replace "Exec=hase" "Exec=$out/bin/hase"
+  '';
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+  meta = with lib; {
+    description = "Hase is an open source gravity based artillery shooter. It is similar to Worms, Hedgewars or artillery, but the gravity force and direction depends on the mass nearby. It is optimized for mobile game consoles like the GP2X, Open Pandora or GCW Zero";
+    homepage = "http://ziz.gp2x.de/hase/";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ colinsane ];
+    platforms = platforms.linux;
```
```suggestion
    license = lib.licenses.gpl3Plus;
    maintainers = with lib.maintainers; [ colinsane ];
    platforms = lib.platforms.linux;
```

uninsane replied:
for the license (gpl3 -> gpl3Plus): i was going off <https://github.com/theZiz/hase/blob/master/LICENSE>.

it doesn't make any mention of gpl3Only v.s. gpl3Plus, and i'm not knowledgeable enough to know which one is appropriate. i'll apply this suggestion though -- i guess when a GPL3 license doesn't explicitly say otherwise, GPL3+ is assumed?

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
+, pkg-config
+, stdenv
```
```suggestion
```

AndersonTorres commented on pkgs/games/hase/default.nix:
```diff
@@ -0,0 +1,51 @@
+{ lib
+, fetchFromGitHub
```
```suggestion
, stdenv
, fetchFromGitHub
```

uninsane commented:
@AndersonTorres feedback applied, except for changing `propagatedBuildInputs` to `buildInputs` as that breaks the Hase build.

AndersonTorres commented on pkgs/top-level/all-packages.nix:
```diff
@@ -23552,6 +23552,8 @@ with pkgs;
 
   spaceship-prompt = callPackage ../shells/zsh/spaceship-prompt {};
 
+  sparrow3d = callPackage ../development/libraries/sparrow3d {};
```
Split sparrow3d into its own commit (no need to open another PR).

In case of doubts, use this tutorial:
https://stackoverflow.com/questions/6217156/break-a-previous-commit-into-multiple-commits

uninsane replied:
okay, i've split the commit

uninsane committed:
sparrow3d: init at unstable-2020-10-06

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

uninsane committed:
hase: init at unstable-2020-10-06

project page: <http://ziz.gp2x.de/hase/index.htm>

"Hase is an open source (GPL) gravity based artillery shooter. It is similar to Worms, Hedgewars or artillery, but the gravity force and direction depends on the mass nearby."

Co-authored-by: Anderson Torres <torres.anderson.85@protonmail.com>

AndersonTorres commented on pkgs/development/libraries/sparrow3d/default.nix:
```diff
@@ -0,0 +1,99 @@
+
+    runHook postCheck
+  '';
+
```
```suggestion
```

AndersonTorres reviewed (approved):
Besides LGTM
